toolkit/governed-map/pallet/src/lib.rs

```diff
@@ -13,16 +13,33 @@
 //! ### Defining size limits
 //!
 //! Before adding the pallet to your runtime, first decide on the limits for the data that it will process:
-//! - `MaxChanges`: the maximum number of changes that will be processed in one inherent invocation.
-//!                 This number should be high enough to guarantee the inherent will always have the
-//!                 capacity required to process incoming changes. Setting this limit above the expected
-//!                 number of keys in use is a safe option.
-//! - `MaxKeyLength`: maximum length of keys used. Be warned that if a key is set in the Governed Map
-//!                   that exceeds this lenght limit, this pallet's inherent will fail and stall block
-//!                   production, with the only recovery path being removal of this key so it is no longer
-//!                   in the change set.
-//! - `MaxValueLength`: maximum length of the value under a key. Same considerations as for `MaxKeyLength`
-//!                     apply.
+//!
+//! #### `MaxChanges`
+//! The maximum number of changes that will be processed in one inherent invocation. Changes
+//! are understood as the diff between previously stored mappings and the currently observed
+//! ones, as opposed to raw on-chain events altering those mappings.
+//!
+//! **Important**: This number must be high enough to guarantee the inherent will always have
+//! the capacity required to process incoming changes. If the number of changes exceeds this
+//! limit, [TooManyChanges][InherentError::TooManyChanges] error will be raised stalling block production.
+//!
+//! If this error occurs on a live chain, then the only way of fixing it is to change the
+//! mappings on Cardano close enough to the last state registered in the pallet to bring the
+//! change count below the limit.
+//! Setting this limit above the expected number of keys in use is a safe option.
```

Comment:
I'm wondering about truncating the changes and leaning towards inconsistent state.
Let stall happen, even immediate reaction requires Cardano blocks to get stable before PC will continue block production.